src/instantsend/instantsend.h

```diff
@@ -46,13 +45,12 @@ struct PendingISLockFromPeer {
 
 struct PendingState {
     bool m_pending_work{false};
-    std::vector<std::pair<NodeId, MessageProcessingResult>> m_peer_activity{};
+    std::vector<std::pair<uint256, PendingISLockFromPeer>> m_pending_is;
```

Comment:
why not just `std::vector<PendingISLockFromPeer>`

Reply:
Because it is used here:
```
auto [more_work, locks] = m_is_manager.FetchPendingLocks();                              
ProcessPendingISLocks(std::move(locks));
```
->
```
void NetInstantSend::ProcessPendingISLocks(std::vector<std::pair<uint256, instantsend::PendingISLockFromPeer>>&& locks_to_process)
{
...
        for (auto& p : locks_to_process) {
            if (bad_is_locks.contains(p.first)) { <---- here
                still_pending.push_back(std::move(p));
            }
        }
```


Otherwise need to recalculate hash here; and one more time inside `NetInstantSend::ProcessPendingInstantSendLocks`